scripts/build/Jenkins/Jenkinsfile

```diff
@@ -384,6 +358,208 @@ def HandleDriveMount(String snapshot, String repositoryName, String projectName,
+    def gemsExist = false
+    def templatesExist = false
+    def restrictedExist = false
+    def engineRepo = false
+    def projectRepo = false
+    def gemRepo = false
+    def templateRepo = false
+    def restrictedRepo = false
+
+    dir(repo.NAME) {
+        if(fileExists("Engines")) {
+            enginesExist = true
+        }
+        if(fileExists("Projects")) {
+            projectsExist = true
+        }
+        if(fileExists("Gems")) {
+            gemsExist = true
+        }
+        if(fileExists("Templates")) {
+            templatesExist = true
+        }
+        if(fileExists("Restricted")) {
+            restrictedExist = true
+        }
+
+        //If this is not a standard composite repo, then it wont have any composite dirs, then we know
+        //its a singular object repo, see what kind it is.
+        if(!enginesExist && !projectsExist && !gemsExist && !templatesExist && !restrictedExist) {
+            if(fileExists("engine.json")) {
+                engineRepo = true
+            }
+            if(fileExists("project.json")) {
+                projectRepo = true
+            }
+            if(fileExists("gem.json")) {
+                gemRepo = true
+            }
+            if(fileExists("template.json")) {
+                templateRepo = true
+            }
+            if(fileExists("restricted.json")) {
+                restrictedRepo = true
+            }
+        }
+    }
+
+    dir(ENGINE_REPOSITORY_NAME) {
+        //register all the objects in this repo
+        //if it is a composite repo then use the --all-<type>-path to register all of them
+        if(enginesExist) {
+            PlatformSh("scripts/o3de${ext} register --all-engines-path ${workspace}/${repoName}/Engines || ver>nul", "Registering all engines in ${workspace}/${repoName}/Engines")
+        }
+        if(projectsExist) {
+            PlatformSh("scripts/o3de${ext} register --all-projects-path ${workspace}/${repoName}/Projects || ver>nul", "Registering all projects in ${workspace}/${repoName}/Projects")
+        }
+        if(gemsExist) {
+            PlatformSh("scripts/o3de${ext} register --all-gems-path ${workspace}/${repoName}/Gems || ver>nul", "Registering all gems in ${workspace}/${repoName}/Gems")
+        }
+        if(templatesExist) {
+            PlatformSh("scripts/o3de${ext} register --all-templates-path ${workspace}/${repoName}/Templates || ver>nul", "Registering all templates in ${workspace}/${repoName}/Templates")
+        }
+        if(restrictedExist) {
+            PlatformSh("scripts/o3de${ext} register --all-restricted-path ${workspace}/${repoName}/Restricted || ver>nul", "Registering all restricted in ${workspace}/${repoName}/Restricted")
+        }
+        //enable all gems in this repo
+        if(gemsExist) {
+            PlatformSh("scripts/o3de${ext} enable-gem --all-gem-paths ${workspace}/${repoName}/Gems --project-name AutomatedTesting || ver>nul", "Enabling all gems in AutomatedTesting")
```

Comment:
Can this `enable-gem` command be combined with the `if(gemsExist)` block above?

Reply:
I don't think so. We want the scripts to run from dir(ENGINE_REPOSITORY_NAME) but the decision to run is based on if a different folder/file exists. The determination of if a file or folder exists is the fileExists() function, and it depends on the current folder so I have to be in that dir to make the determination but I need to be in the engine dir in order to run the command, so i have to save off the determination, change dir, execute based on the determination. I don't know of any way to call fileExists() or PlatformSh() from a dir outside the current dir... This logic is simple enough and we know this works, so we should just keep it.

Reply:
I'll try it

Reply:
I'm fine to leave as-is if it keeps the jenkinsfile logic simplified.

Reply:
updated: needs to be tested though.

Reply:
done